open-hackathon-server/src/hackathon/hack/host_server_manager.py

```diff
@@ -190,6 +209,142 @@ def create_docker_host_vm(self, hackathon_id):
+        """
+        create a docker host DB object for a hackathon and insert record into the database.
+        param-"args" contain all necessary infos to new a docker_host
+
+        :param hackathon_id: the id of hackathon in DB
+        :type hackathon_id: Integer
+
+        :return: return True if succeed, otherwise return False
+        :rtype: bool
+        """
+
+        try:
+            ping_url = 'http://%s:%d/_ping' % (args.public_dns, int(args.public_docker_api_port))
+            req = requests.get(ping_url, timeout=5)
+            if req.status_code == 200 and req.content == DockerPingResult.OK:
+                args['state'] = DockerHostServerStatus.DOCKER_READY
+            else:
+                args['state'] = DockerHostServerStatus.UNAVAILABLE
+        except Exception as e:
+            self.log.error(e)
+            args['state'] = DockerHostServerStatus.UNAVAILABLE
+
+        self.db.add_object_kwargs(DockerHostServer,
+                                  vm_name = args.vm_name,
+                                  public_dns = args.public_dns,
+                                  public_ip = args.public_ip,
+                                  public_docker_api_port = args.public_docker_api_port,
+                                  private_ip = args.private_ip,
+                                  private_docker_api_port = args.private_docker_api_port,
+                                  container_count = 0,
+                                  container_max_count = args.container_max_count,
+                                  is_auto = AzureVMStartMethod.MANUAL,
+                                  state = args.state,
+                                  disabled = args.disabled,
+                                  create_time = self.util.get_now(),
+                                  update_time = self.util.get_now(),
+                                  hackathon_id = hackathon_id)
+        return ok()
+
+    def get_and_check_host_server(self, hackathon_id, host_server_id):
+        """
+        first get the docker host DB object for a hackathon.
+        and check whether the container_count is correct, if not, update this value in the database.
+
+        :param hackathon_id: the id of hackathon in DB
+        :type hackathon_id: Integer
+
+        :param host_server_id: the id of host_server in DB
+        :type host_server_id: Integer
+
+        :return: A object of docker_host_server
+        :rtype: DockerHostServer object or None
+        """
+        vm = self.db.find_first_object_by(DockerHostServer, id=host_server_id, hackathon_id=hackathon_id)
+        if vm is None:
+            self.log.warn('get docker_host fail, not find hostserver_id:' + host_server_id
+                          + 'hackathon_id:' + hackathon_id)
+            return None
+
+        vm_data = vm.dic()
+        try:
+            containers_url = 'http://%s:%d/containers/json' % (vm.public_dns, int(vm.public_docker_api_port))
```

Comment:
again. We don't want to know docker api details other than `hosted_docker.py`. This class is designed for the host server management not docker management.